data_manager_select_index_by_path/data_manager/test_path_name_value_key_manager.py

```diff
@@ -18,83 +18,56 @@
 test_data = Path(__file__).parent.parent / Path("test-data")
 
 
+@pytest.fixture
+def temp_json_path():
+    with tempfile.NamedTemporaryFile(suffix=".json",
+                                     dir=TEST_OUTPUT_DIR) as tmp:
+        path = tmp.name
+        # [1] Needed. mkstemp returns a tuple.
+        # The second value is the absolute path
+    yield path
+    os.remove(path)
+
+
 def test_application():
-    output_path = tempfile.mkstemp(".json", "fasta_indexes", TEST_OUTPUT_DIR)[
-        1]
-    # [1] Needed. mkstemp returns a tuple.
-    # The second value is the absolute path
-    os.remove(output_path)  # File may not exist.
-    index_path = test_data / Path("fasta_indexes/EboVir3.fa")
-    sys.argv = ['',
-                "--path", str(index_path),
-                "--data_table_name", "fasta_indexes",
-                "--json_output_file", str(output_path)
-                ]
-    main()
-    data_manager_dict = \
-        json.load(Path(output_path).open())["data_tables"]['fasta_indexes'][0]
-    assert (data_manager_dict['path'] == str(index_path))
-    assert (data_manager_dict['name'] == "EboVir3")
-    assert (data_manager_dict['value'] == "EboVir3")
-    assert (data_manager_dict['dbkey'] == "EboVir3")
+    for output_path in temp_json_path():
```

Comment:
To use the fixtures, you're not supposed to call them like that.

Simply use the fixture name as an named argument to your function:

```
def test_application(temp_json_path):
     # do sth with the temp_json_path, no need to loop!
    ....
```

Reply:
Thanks.
Done!